src/Microsoft.Diagnostics.Monitoring.WebApi/IDiagnosticServices.cs

```diff
@@ -11,12 +11,12 @@ namespace Microsoft.Diagnostics.Monitoring.WebApi
     /// Set of services provided by the monitoring tool. These are consumed by
     /// the REST Api.
     /// </summary>
-    internal interface IDiagnosticServices
+    public interface IDiagnosticServices
```

Comment:
I was able to undo the `IDiagnosticServices` visibility changes.

For the remainder... ASP.NET controllers need to be public in order to participate in routing. And since the `InvokeForProcess` methods are protected, they are also effectively public for any deriving class in respect to the WebApi assembly. Thus, the method parameter types need to become public as well.

Reply:
Now using `private protected` on the `InvokeForProcess` methods, which allows the other types to remain `internal`: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private-protected